lib/eigensolve_quda.cpp

```diff
@@ -130,8 +135,13 @@ namespace quda
       if (!eig_solver->hermitian()) errorQuda("Polynomial acceleration not supported with non-Hermitian solver");
     }
 
+    // Cannot solve for imaginary spectrum of hermitian systems
     if (mat.hermitian() && (eig_param->spectrum == QUDA_SPECTRUM_SI_EIG || eig_param->spectrum == QUDA_SPECTRUM_LI_EIG))
       errorQuda("The imaginary spectrum of a Hermitian operator cannot be computed");
+
+    // Cannot compute SVD of non-normal operators
+    if (!eig_param->use_norm_op && eig_param->compute_svd)
```

Comment:
I thought that the point of the SVD was that it exists for any matrix regardless of its property and shape? https://en.wikipedia.org/wiki/Singular_value_decomposition

Reply:
@weinbe2 Yes, that's right, but I haven't coded up a routine that can compute the SVD of a non-hermitain matrix M. Instead, we compute the eigendecomposition of MdagM, which also furnishes the right singular vectors of M:
```
MdagM * X = lambda X
MdagM * X_right = Mdag * sigma * X_left = sigma^2 X_right
``` 
hence the sigma and X_right data can be computed via the (admittedly poorer conditioned) MdagM operator. The X_left vectors are easily identified via
```
M * X_right = sigma * X_left
```
I think it's OK to use the poorer MdagM as we only use the SVD in QUDA to deflate non-hermitian solvers, so the extra time required to identify the SVD via MdagM rather than M is amortised by the solver speed-up.